cni/pkg/plugin/kubernetes.go

```diff
@@ -67,47 +69,43 @@ func getK8sPodInfo(client *kubernetes.Clientset, podName, podNamespace string) (
 	}
 
 	pi := &PodInfo{
-		InitContainers:    make(map[string]struct{}),
-		Containers:        make([]string, len(pod.Spec.Containers)),
+		Containers:        sets.New[string](),
 		Labels:            pod.Labels,
 		Annotations:       pod.Annotations,
 		ProxyEnvironments: make(map[string]string),
 	}
-	for _, initContainer := range pod.Spec.InitContainers {
-		pi.InitContainers[initContainer.Name] = struct{}{}
-	}
-	for containerIdx, container := range pod.Spec.Containers {
-		log.Debugf("Inspecting pod %v/%v container %v", podNamespace, podName, container.Name)
-		pi.Containers[containerIdx] = container.Name
-
-		if container.Name == "istio-proxy" {
+	for _, c := range containers(pod) {
+		pi.Containers.Insert(c.Name)
+		if c.Name == "istio-proxy" {
```

Comment:
```suggestion
		if c.Name == ISTIO_NODEPROXY {
```

meganit: `const` as elsewhere in the package? maybe global to `plugin` package?